app/controllers/failure_modes_controller.rb

```diff
@@ -5,20 +5,20 @@ class FailureModesController < ApplicationController
   def update
     @failure_mode.update(failure_mode_params)
     @fmea = @failure_mode.function.fmea
-    redirect_to edit_fmea_path(@fmea, anchor: card_id(@failure_mode))
+    # redirect to where the failure mode was added
+    redirect_to edit_fmea_path(@fmea, anchor: card_id(@failure_mode.function))
   end
 
   def create
-    @failure_mode = FailureMode.create(failure_mode_params)
-    @function = Function.find(params["failure_mode"]["function_id"]) # Why so complicated? Because this was passed with the simple form using the hidden-value. It does not return the object straight up.
-    @failure_mode.function = @function
-    @fmea = @function.fmea
+    @failure_mode = FailureMode.new(failure_mode_params)
 
     if @failure_mode.save
-      redirect_to(edit_fmea_path(@fmea))
+      # redirect to the function where the failure mode was added
+      redirect_to edit_fmea_path(@failure_mode.function.fmea, anchor: card_id(@failure_mode.function))
     else
-      @fmeas = Fmea.all
-      render action: :index
```

Comment:
when i tested this code it didnt seem to work and was also inconsistent with the functions controller, therefore i removed and added the code below. The user will get a flash alert with any error messages that are recorded.